src/plugins/intel_gpu/src/graph/fully_connected.cpp

```diff
@@ -246,14 +249,36 @@ kernel_impl_params fully_connected_inst::get_fake_aligned_params(kernel_impl_par
                 can_apply_fake_alignment = false;
                 break;
             }
+            auto fuse_op_input_shape = fused_op_input_layout.get_shape();
+
             // Check fused desc's input has full tensor, then do not fake alignment
-            if (orig_output_layout.get_shape() == fused_op_input_layout.get_shape()) {
+            if (orig_output_layout.get_shape() == fuse_op_input_shape) {
                 can_apply_fake_alignment = false;
                 break;
             }
+
+            if (fuse_op_input_shape.size() > 2) {
+                auto fuse_op_batch_size = std::accumulate(fuse_op_input_shape.begin(),
+                                                          fuse_op_input_shape.end() - 1,
+                                                          size_t{1},
+                                                          std::multiplies<size_t>());
+                if (fuse_op_batch_size > 1) {
+                    GPU_DEBUG_TRACE_DETAIL << "unable to apply fake-alignment: "
+                                           << orig_impl_param.typed_desc<fully_connected>()->id
+                                           << " " << input_shape
+                                           << " + " << fuse_op_input_shape << std::endl;
+                    can_apply_fake_alignment = false;
```

Comment:
NOTE: I think we can add a reshape for this case in the future

Reply:
Oh, I'll try it in the future as you reviewed. Thank you!